AK/String.cpp

```diff
@@ -254,6 +254,22 @@ ErrorOr<String> String::substring_from_byte_offset(size_t start) const
     return substring_from_byte_offset(start, bytes_as_string_view().length() - start);
 }
 
+ErrorOr<String> String::substring_from_code_unit_offset(size_t offset, size_t count) const
+{
+    auto const utf16_data = TRY(utf8_to_utf16(bytes_as_string_view()));
+    Utf16View const utf16_view { utf16_data };
+
+    return TRY(utf16_view.substring_view(offset, count).to_utf8());
+}
+
+ErrorOr<String> String::substring_from_code_unit_offset(size_t offset) const
```

Comment:
The spec doesn't explicitly say "utf16 code unit" since it internally represents strings as utf16, but we should explicitly call it out. May need to also pass through a flag to allow for invalid UTF8 as well

Reply:
We should also leave a FIXME saying that this isn't efficient - at some stage it looks likely that we need to implement utf16 caching on String - like CharacterData we should call out that this will need to be made more efficient